### DISCUSSION TITLE
DataGrid - issue with DbContext when using IQueryable and sorting quickly?

### DISCUSSION THREAD

**Discussion by leetheman** [microsoft/fluentui-blazor#763] [Q&A]
Hi - me again.

I've got a DbContext that is scoped to the component (I've implemented the IDisposable interface) and this works really well for the IQueryable DataGrid - except if you try to sort quickly.  If you do, you sometimes end up with the "a second operation cannot be started" error.  Is there a way to prevent the user from performing another sort until the initial sort operation is complete?

Cheers

Lee

**Comment by vnbaaij** [comment]
I don't have an answer for you here. Perhaps following the links shown in the EF paragraph at https://aspnet.github.io/quickgridsamples/datasources can help you moe forward with this
